docs/backend-api/api-methods.md

```diff
@@ -193,19 +197,262 @@ You will need to generate new token after the first one expires.
 
   OR
 
-  * **Code:** `401 UNAUTHORIZED` <br />
+  * **Code** `401 UNAUTHORIZED` <br />
 
   OR
 
-  * **Code:** `422 Unprocessable Entity` <br />
-    **Content:**
+  * **Code** `422 Unprocessable Entity` <br />
+    **Content**
     ```
     {
         "code": 422,
         "message": "There are no products on provided page."
     }
     ```
 
-* **Sample Call:**
+* **Sample Call**
 
     `http://127.0.0.1:8000/api/v1/products?uuids[]=877a07d5-a276-49f8-9b73-6ac0edef83be&uuids[]=f564da76-6d51-4ecd-b004-895c8019a235x   `
+
+**Add Product**
+----
+
+* **URL**
+
+  `/api/v1/products`
+
+* **Method**
+
+  `POST`
+
+* **Request Body**
+
+  * **Content**
+    ```
+    {
+      "name": {
+          "en": "22\" Sencor SLE 22F46DM4 HELLO KITTY",
+          "cs": "22\" Sencor SLE 22F46DM4 HELLO KITTY"
+      },
+      "hidden": false,
+      "sellingDenied": false,
+      "sellingFrom": "2000-01-16T00:00:00+00:00",
+      "sellingTo": "2100-01-16T00:00:00+00:00",
+      "catnum": "9177759",
+      "ean": "8845781245930",
+      "partno": "SLE 22F46DM4",
+      "shortDescription": {
+          "1": "Television LED, 55 cm diagonal ...",
+          "2": "Sencor SLE 22F46DM4 Hello Kitty je ..."
+      },
+      "longDescription": {
+          "1": "Television LED, 55 cm diagonal ...",
+          "2": "<p><strong>Sencor SLE 22F46DM4 ..."
+      }
+    }
+    ```
+
+* **Success Response**
+
+  * **Code** `201`  
+    **Headers**
+    `Location: http://127.0.0.1:8000/api/v1/products/7a34a8cf-8f3b-405d-9662-f20d983cfc8a`
+
+* **Error Response**
+
+  * **Code** `401 UNAUTHORIZED`
+
+  OR
+
+  * **Code** `400 Bad Request`  
+    **Content**
+    ```
+    {
+        "message": "Provided data did not pass validation",
+        "errors": [
+            "[sellingTo]" => "The value \"12.1.2019\" is not a valid DateTime::ATOM format.",
```

Comment:
Why square brackets? Seems not important to me.

Reply:
That's because the Symfony validator produces it in this form and we didn't have a reason for removing them.

Nested errors have keys for example `[shortDescription][1]`, so "just removing" would be eventually complicated.

Do you think I shall change the format in some way? If so, please suggest one and keep in mind nested errors

Reply:
I liked the format initially introduced in documentation:
```json
{
  "errors": {
    "sellingTo":  "The value \"12.1.2019\" is not a valid DateTime::ATOM format.",
    "shortDescription.1": "Bad word found."
  }
}
```

or nested structure 
```json
{
  "errors": {
    "sellingTo":  "The value \"12.1.2019\" is not a valid DateTime::ATOM format.",
    "shortDescription": {
      "1": "Bad word found."
    }
  }
}
```
and the main argument to handle this: **Returned response is not a valid JSON right now.**

Reply:
I can only agree. Changed to dot notation

Reply:
I still struggle with the format of the response.
```
{
    "message": "Provided data did not pass validation",
    "errors": [
        "sellingTo" => "The value \"12.1.2019\" is not a valid DateTime::ATOM format.",
        "longDescription.1" => "The value 345 is not a valid string.",
    ]
}
```

is not valid JSON and I believe it should be

Reply:
how did you get this response?

I just tried to POST with errors, and I got
```json
{
    "code": 400,
    "message": "Provided data did not pass validation",
    "errors": {
        "name.en": "The value \"22\" ......\" cannot be longer then 255 characters",
        "extId": "This field was not expected."
    }
}
```

And this is a valid JSON

Reply:
Ou I didn't test it live... The invalid format is in documentation `api-methods.md` and I assumed it's the same as the return value from the code :)